core/api/src/servers/middlewares/scope.ts

```diff
@@ -0,0 +1,79 @@
+import { GraphQLResolveInfo, GraphQLFieldResolver } from "graphql"
+
+import ScopesOauth2 from "@/domain/authorization"
+import { AuthorizationError } from "@/domain/errors"
+import { mapError } from "@/graphql/error-map"
+import { mutationFields } from "@/graphql/public"
+
+const readTransactionsAuthorize = async (
+  resolve: GraphQLFieldResolver<unknown, GraphQLPublicContextAuth>,
+  parent: unknown,
+  args: unknown,
+  context: GraphQLPublicContextAuth,
+  info: GraphQLResolveInfo,
+) => {
+  const scope = context.scope
+  const appId = context.appId
+
+  // not a delegated token
+  if (appId === undefined || appId === "") {
+    return resolve(parent, args, context, info)
+  }
+
+  if (scope === undefined) {
```

Comment:
this will never hit because of line 19.

Reply:
line 19 is a check on appId, not scope?

here the idea is that if this is a hydra/delegation request, I want scope to exist.

Reply:
Whoops misread